### PR TITLE
BPTL Dashboard: Add homecollection APIs (Part 1)

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
This PR addresses following issue:
https://github.com/episphere/connect/issues/771

Related PR:
https://github.com/episphere/biospecimen/pull/609

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -2026,6 +2047,34 @@ const getKitAssemblyData = async () => {
...
+        .where('827220437', '==', 125001209)
+        .get();
+        return snapShot.docs.map(document => processParticipantData(document.data()));
+    } catch (error) {
+        return new Error(error);
+    }
+}
+
+const processParticipantData = (record) => {
+    const processedRecord = {
```
Comment (jhflorey) [inline utils/firestore.js:2064]: NIT: We can return data directly without declaring `processedRecord`

**Review by jhflorey — APPROVED** [review]

File: utils/biospecimen.js [hunk 2]
```diff
@@ -623,16 +623,24 @@ const biospecimenAPIs = async (req, res) => {
...
-        requestData.id = uuid();
-        requestData.timeStamp = currentDate;
         const { addKitAssemblyData } = require('./firestore');
         const response = await addKitAssemblyData(requestData);
         if(!response) return res.status(404).json(getResponseJSON('ERROR!', 404));
         return res.status(200).json({message: `Success!`, code:200})
     }
 
+    else if(api == 'updateKitData'){
+        if(req.method !== 'POST') {
```
Comment (Gbarra9) [inline utils/biospecimen.js:633]:
```suggestion
        if (req.method !== 'POST') {
```

nit: spacing after `if` keyword

**Review by Gbarra9 — APPROVED** [review]

**Review by jhflorey — APPROVED** [review]

**Review by Gbarra9 — APPROVED** [review]

File: utils/firestore.js [hunk 3]
```diff
@@ -2014,6 +2013,38 @@ const addKitAssemblyData = async (data) => {
...
+const updateKitAssemblyData = async (data) => {
+    try {
+        const snapShot = await db.collection('kitAssembly').where('687158491', '==', data['687158491']).get();
+
+        if (snapShot.empty) return false
+        const docId = snapShot.docs[0].id;
+ 
+        await db.collection('kitAssembly').doc(docId).update({
+            '194252513': data['194252513'],
+            '259846815': data['259846815'],
```
Comment (anthonypetersen) [inline utils/firestore.js:2025]: might be helpful if we used constants here for readability

File: utils/firestore.js [hunk 4]
```diff
@@ -2014,6 +2013,38 @@ const addKitAssemblyData = async (data) => {
...
+    catch(error){
+        console.error(error);
+        return new Error(error);
+    }
+}
+
+const checkCollectionUniqueness = async (id) => {
+    try {
+        const snapShot = await db.collection('kitAssembly').where('259846815', '==', id).get();
+        if (snapShot.docs.length === 0) { return true }
```
Comment (anthonypetersen) [inline utils/firestore.js:2041]: does `snapShot.docs.length === 0` do the same thing as `snapShot.empty` on line 2020? If so, I'd just suggest picking one and using it in both places for consistency

File: utils/firestore.js [hunk 5]
```diff
@@ -2026,6 +2057,34 @@ const getKitAssemblyData = async () => {
...
+        return new Error(error);
+    }
+}
+
+const processParticipantData = (record) => {
+    const processedRecord = {
+        first_name: record['399159511'],
+        last_name: record['996038075'],
+        address_1: record['521824358'],
+        address_2: record['442166669'] || '',
```
Comment (anthonypetersen) [inline utils/firestore.js:2078]: constants might help here too

**Review by anthonypetersen — APPROVED** [review]
few minor comments, looks good regardless

**Pull request merged**